contracts/TaskBook.sol

```diff
@@ -5,83 +5,158 @@ import './AccountManager.sol';
 contract TaskBook is AccountManager {
 
 	uint private numTasks = 0;
-	uint constant maxSolvers = 10;
-	uint constant maxChallengers = 10;
+	uint private forcedErrorThreshold = 42;
 
 	event TaskCreated(uint taskID, uint minDeposit, uint blockNumber);
 	event SolverSelected(uint indexed taskID, address solver, bytes32 taskData, uint minDeposit);
-	event SolutionCommitted(uint taskID, uint minDeposit, bytes32 taskData, address solver);
+	event SolutionsCommitted(uint taskID, uint minDeposit, bytes32 taskData, address solver);
+	event SolutionRevealed(uint taskID, uint randomBits);
+	event TaskStateChange(uint taskID, uint state);
 
-	mapping(uint => Task) private tasks;
-	mapping(address => mapping(uint => bytes32)) private solverRandomBitsHash;
-	mapping(address => mapping(uint => bytes32)) private solutions;
-	mapping(address => mapping(uint => bytes32)) private verifierIntent;
+	struct Challenge {
+		address challenger;
+		bytes32 intentHash;
+	}
 
 	struct Task {
 		address owner;
-		address[] solvers;
 		address selectedSolver;
 		uint minDeposit;
 		bytes32 taskData;
-		uint numSolvers;
-		address[] challengers;
-		uint numChallengers;
+		mapping(address => bytes32) challenges;
+		uint state;
```

Comment:
Change `state` to an enum?

Reply:
Yeah that would be more expressive.

Reply:
Turns out you can't convert uint -> enum, so this doesn't actually work for us without an ugly solution.

Reply:
Shouldn't converting from uint to enum be just `E(n)` if `E` is the enum type and `n` is an int  variable?

Reply:
Maybe something like this would work:

```
enum Stages {
  TaskCreated,
  SolverBidsCollected,
  SolverSelected,
  SolverSubmittedSolutions,
  SolverPickedSolutions,
  ChallengeBidsCollected,
  ChallengeBidsRevealed,
  ChallengerSelected
}

struct Task {
  Stages stage;
  address taskGiver;
  //...
}

// Transition the Task's state to the next.
function taskToNextStage(uint taskID) returns (bool) {
  Task t = tasks[taskID];
  t.stage = Stages(uint(t.stage) + 1);
}

```

Reply:
Think it would be fine if you merged this PR, and then we attempted the switch to enum after (just to keep the ball moving).